pipeline/frontend/yaml/compiler/convert.go

```diff
@@ -80,11 +79,11 @@ func (c *Compiler) createProcess(container *yaml_types.Container, stepType backe
 
 	environment["CI_WORKSPACE"] = path.Join(c.base, c.path)
 
-	if stepType == backend_types.StepTypeService || container.Detached {
-		detached = true
+	if container.Detached {
```

Comment:
until we dont have detatched removed from the frontend we should not remove that!!

else a user can create two services with the same hostname

Reply:
well i have a solution: prefix the step name with the uuid ... this way you can not guess the hostname (as it was with non service-steps that are detached) and they cant clash